crates/shared/src/http_solver/model.rs

```diff
@@ -325,6 +326,78 @@ pub struct ExecutionPlanCoordinatesModel {
     pub position: u32,
 }
 
+/// The result a given solver achieved in the auction
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub enum AuctionResult {
+    /// Solution was valid and was ranked at the given place
+    /// Rank 1 means the solver won the competition
+    Ranked(usize),
+
+    /// Solution was invalid for some reason
+    Rejected(SolverRejectionReason),
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub enum SolverRejectionReason {
+    /// The solver didn't return a successful response
+    RunError(SolverRunError),
+
+    /// The solution candidate didn't include any user orders
+    NoUserOrders,
+
+    /// The solution candidate didn't include any mature user orders
+    NoMatureOrders,
+
+    /// The solution violated a price constraint (ie. max deviation to external price vector)
+    PriceViolation,
```

Comment:
The maturity requirement is currently implemented in an unintuitive way and the price violation depends on a configuration we set (also it would be helpful to give them the token that had the price violation).
Having some public documentation for those errors or very descriptive error messages including the current relevant parameters would probably help.

Reply:
I agree this data can be richer and expect to receive further requests for requirement change from solvers (I want them to ask for us to provide them something they will surely use rather than adding a lot of stuff that might not even use). But for the first version (and since we want to unblock Seasolver with this PR) I think we can merge this PR as is.